### DISCUSSION TITLE
Why can't I import plugin from OpenAPI spec?

### DISCUSSION THREAD

**Discussion by EMadd2015** [microsoft/semantic-kernel#8527] [Q&A] [answered]
I need a flexible way to define the input and output of functions.
I like the way a plugin can be added using native code, since it supports a schema for complex objects, including enumerated values. However, C# code is not flexible enough for my use case, and I would rather not use reflection.
The alternative given in the documentation is to use an OpenAPI spec, but according to my Intellisense in Visual Studio, there is no namespace or class named 'Plugins' in 'SemanticKernel,' and therefore no 'OpenApi,' nor a 'ImportPluginFromOpenApi.' I'm not sure how this could possibly be, as I see these things in src, and in documentation. The version of Semantic Kernel I am using is 1.18.2.
The other alternative, adding a plugin from prompt, or prompt directory, only seems to support string or integer parameters, and no schema for complex objects.

In short, how do I get all the features of plugins from native code, with all the flexibility of JSON?

**Comment by moonbox3** [comment]
@markwallace-microsoft / @SergeyMenshykh for visibility.

**Comment by EMadd2015** [comment, marked as answer]
I figured out my issue with importing plugins from OpenAPI specs...
I didn't have the checkbox in Visual Studio's Nuget Package Manager checked to include prereleases.
I didn't realize that the Plugins.OpenApi package was an experimental feature in alpha. I don't recall seeing anything in the documentation suggesting that this feature was experimental.

**Reply by SergeyMenshykh** [reply]
Hi @EMadd2015, thanks for reporting the issue and providing a solution! We have the OpenAPI package graduation in our roadmap and plan to start working in that direction soon.

**Reply by parthInsight** [reply]
I was having the same issue and it got resolved by checking the Include Pre release option . But now When i try to import an OpenAPI spec file in the Semantic Kernel it throws me an error "{Microsoft.SemanticKernel.HttpOperationException}". Any  fixes for this ?
